### PR TITLE
14 create view for signup

### PR DESCRIPTION
**Pull request by xuvea**
Created the view of the Signup page. No functionality yet besides some placeholder links that will be replaced once there is a forgot password portal. Functionality will be implemented later.

Some screenshots of what was implemented:
pcdark
pclight
inputtest
popup

### CODE REVIEW

**Review by juroc95 — COMMENTED** [review]
- Firebase provides "sign in with google" button
- There are some conflicts that must be resolved in package-lock.json
- Can we change "CONFIRM" to "Sign up"?

Good work!

**Comment by xuvea** [comment]
I will change these. For some reason my it keeps changing "lockfileVersion" to 2 instead of 1. I need to look into this.

File: src/pages/signup/Signup.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                  size="default"
+                  class="gbutton"
+                >
+                  <IonIcon icon={logoGoogle}></IonIcon>
+                  Google Alternate Sign In
+                </IonButton>
+              </IonItem>
+              {/*Need hyperlink for the forgot password once implemented*/}
+              <IonCardContent class="fpass" color="light">
+                <a href="https://www.youtube.com/watch?v=dQw4w9WgXcQ&ab_channel=RickAstley">
```
Comment (bailee-crandley) [inline src/pages/signup/Signup.tsx:108]: Noice

**Comment by xuvea** [comment]
image

File: src/pages/signup/Signup.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+import "./Signup.css";
+
+const Signup: React.FC = () => {
+  const [presentAlert] = useIonAlert();
+
+  function validateEmail() {
+    presentAlert({
+      header: "Alert",
+      cssClass: "center-alert",
+      subHeader: "There is already an existing account under this email",
```
Comment (bailee-crandley) [inline src/pages/signup/Signup.tsx:28]: This might be a no-no from a security perspective. I was told never to do this when I did my OWASP security training but it seems like there's some discussion around it: https://security.stackexchange.com/questions/158075/is-it-unsafe-to-show-message-that-username-account-does-not-exist-at-login 
TLDR; A user should not be able to find out if an account exists or not without signing in. I don't know how to avoid this.

Comment (abluescarab) [reply src/pages/signup/Signup.tsx:28]: Most websites just say "That username and password combination does not exist" or something similar, then rely solely on the "forgot password" link to let someone reset their password. It also doesn't indicate on the forgot password form whether an email exists or not, just says that an email has been sent.

Comment (bailee-crandley) [reply src/pages/signup/Signup.tsx:28]: @abluescarab That's a good point but we're not talking about forgotten passwords. This is the scenario where someone already has an account and they try to make another one with the same email.

**Comment by bailee-crandley** [comment]
I have the questions that they would normally ask during registration now so I'm going to create a new story for adding that as well. It could either be on this same sign up page or it could be on a page directly following this. Either way, the sponsors requested that these questions are asked during the signup process and users don't gain access to the rest of the site until they're answered.

**Review by bailee-crandley — APPROVED** [review]

**Comment by bailee-crandley** [comment]
Ready to merge. Blocked by lockfile, npm needs to be backed up? Might already be merged. Check if changes are on main.
